### ISSUE TITLE
[bug] menu : Uncaught Error : ViewDestroyedError : Attempt to use a destroyed view : detectChanges

### ISSUE THREAD

**Issue by Neaco** [telerik/kendo-angular#1586]
## I'm submitting a...

* Bug report 

## Current behavior

I have a nested menu with 4 levels. Regularly, when I hover on the 3rd submenu to open the last level, I get an error where the other submenus close and only the last level stays. It won't disappear until I reload the page.

### The bug : 
![menu bug](https://user-images.githubusercontent.com/22844693/40907548-cb58e3b2-6816-11e8-9f04-d7d61884bb9f.gif)

### The chrome developper tools error : 
https://i.imgur.com/kjBldnx.png
![image](https://user-images.githubusercontent.com/22844693/40907624-031f8f6c-6817-11e8-801e-29ea60213442.png)

## Minimal reproduction of the problem with instructions

I couldn't find an easy set up to reproduce that bug on plunk.r. If needed I'll try to recreate it.

The template : 
```html
<div id="menu-container">
  <kendo-menu [items]="windowList"
              [openOnClick]="true"
              [hoverDelay]="500">
    <ng-template kendoMenuItemTemplate
                 let-item="item">
      <span (click)="onMenuClick(item,$event)"> {{item.text}}{{item.description}}{{item.report}} {{item.date | date : mediumDate}} {{item?.instance?.title | titlecase }}</span>
      <!-- <span *ngIf="item?.instance?.title"
            (click)="onWindowBarClick(item)"> {{item.instance.title | titlecase }}</span> -->
      <!-- <span *ngIf="item?.date"> {{item.date | date:mediumDate }}</span> -->
    </ng-template>
  </kendo-menu>
</div>
```

## Environment

Package versions:

```
+-- @angular-devkit/build-angular@0.6.7
+-- @angular/animations@6.0.3
+-- @angular/cli@6.0.7
+-- @angular/common@6.0.3
+-- @angular/compiler@6.0.3
+-- @angular/compiler-cli@6.0.3
+-- @angular/core@6.0.3
+-- @angular/forms@6.0.3
+-- @angular/http@6.0.3
+-- @angular/language-service@6.0.3
+-- @angular/platform-browser@6.0.3
+-- @angular/platform-browser-dynamic@6.0.3
+-- @angular/router@6.0.3
+-- @compodoc/compodoc@1.1.3
+-- @progress/kendo-angular-buttons@4.1.1
+-- @progress/kendo-angular-dateinputs@3.4.0
+-- @progress/kendo-angular-dialog@3.6.1
+-- @progress/kendo-angular-dropdowns@3.0.1
+-- @progress/kendo-angular-excel-export@2.1.0
+-- @progress/kendo-angular-grid@3.4.0
+-- @progress/kendo-angular-inputs@3.1.2
+-- @progress/kendo-angular-intl@1.4.1
+-- @progress/kendo-angular-l10n@1.2.0
+-- @progress/kendo-angular-layout@3.1.0
+-- @progress/kendo-angular-menu@0.4.0
+-- @progress/kendo-angular-pdf-export@1.1.0
+-- @progress/kendo-angular-toolbar@0.2.1
+-- @progress/kendo-angular-treeview@2.2.0
+-- @progress/kendo-data-query@1.3.1
+-- @progress/kendo-date-math@1.2.0
+-- @progress/kendo-drawing@1.5.6
+-- @progress/kendo-theme-bootstrap@2.13.5
+-- @progress/kendo-theme-default@2.53.2
+-- @types/jasmine@2.8.8
+-- @types/jasminewd2@2.0.3
+-- @types/jquery@3.3.2
+-- @types/node@6.0.112
+-- bootstrap@4.0.0
+-- codelyzer@4.3.0
+-- core-js@2.5.7
+-- jasmine-core@2.8.0
+-- jasmine-spec-reporter@4.2.1
+-- UNMET PEER DEPENDENCY jquery@1.9.1 - 3
+-- karma@2.0.2
+-- karma-chrome-launcher@2.2.0
+-- karma-coverage-istanbul-reporter@1.4.3
+-- karma-jasmine@1.1.2
+-- karma-jasmine-html-reporter@0.2.2
+-- UNMET PEER DEPENDENCY popper.js@^1.12.9
+-- protractor@5.1.2
+-- rxjs@6.2.0
+-- rxjs-compat@6.2.0
+-- standard-version@4.4.0
+-- ts-node@4.1.0
+-- tslint@5.9.1
+-- typescript@2.7.2
+-- webpack-bundle-analyzer@2.13.1
`-- zone.js@0.8.26

npm ERR! peer dep missing: jquery@1.9.1 - 3, required by bootstrap@4.0.0
npm ERR! peer dep missing: popper.js@^1.12.9, required by bootstrap@4.0.0
npm ERR! missing: mkdirp@0.5.1, required by node-pre-gyp@0.9.1
npm ERR! missing: minimist@0.0.8, required by mkdirp@0.5.1
npm ERR! missing: minimatch@3.0.4, required by ignore-walk@3.0.1
npm ERR! missing: brace-expansion@1.1.11, required by minimatch@3.0.4
npm ERR! missing: balanced-match@1.0.0, required by brace-expansion@1.1.11
npm ERR! missing: concat-map@0.0.1, required by brace-expansion@1.1.11
npm ERR! missing: console-control-strings@1.1.0, required by npmlog@4.1.2
npm ERR! missing: inherits@2.0.3, required by readable-stream@2.3.6
npm ERR! missing: safe-buffer@5.1.1, required by readable-stream@2.3.6
npm ERR! missing: safe-buffer@5.1.1, required by string_decoder@1.1.1
npm ERR! missing: console-control-strings@1.1.0, required by gauge@2.7.4
npm ERR! missing: string-width@1.0.2, required by gauge@2.7.4
npm ERR! missing: strip-ansi@3.0.1, required by gauge@2.7.4
npm ERR! missing: code-point-at@1.1.0, required by string-width@1.0.2
npm ERR! missing: is-fullwidth-code-point@1.0.0, required by string-width@1.0.2
npm ERR! missing: strip-ansi@3.0.1, required by string-width@1.0.2
npm ERR! missing: number-is-nan@1.0.1, required by is-fullwidth-code-point@1.0.0
npm ERR! missing: ansi-regex@2.1.1, required by strip-ansi@3.0.1
npm ERR! missing: string-width@1.0.2, required by wide-align@1.1.2
npm ERR! missing: inherits@2.0.3, required by glob@7.1.2
npm ERR! missing: minimatch@3.0.4, required by glob@7.1.2
npm ERR! missing: once@1.4.0, required by glob@7.1.2
npm ERR! missing: once@1.4.0, required by inflight@1.0.6
npm ERR! missing: wrappy@1.0.2, required by inflight@1.0.6
npm ERR! missing: wrappy@1.0.2, required by once@1.4.0
npm ERR! missing: minipass@2.2.4, required by tar@4.4.1
npm ERR! missing: mkdirp@0.5.1, required by tar@4.4.1
npm ERR! missing: safe-buffer@5.1.1, required by tar@4.4.1
npm ERR! missing: yallist@3.0.2, required by tar@4.4.1
npm ERR! missing: minipass@2.2.4, required by fs-minipass@1.2.5
npm ERR! missing: safe-buffer@5.1.1, required by minipass@2.2.4
npm ERR! missing: yallist@3.0.2, required by minipass@2.2.4
npm ERR! missing: minipass@2.2.4, required by minizlib@1.1.0
```

Browser:

- Chrome (desktop) version 66.0.3359.181
- Firefox version 53.0.2 (32-bit)

System:
- Node version: v8.10.0  npm v5.6.0
- Platform:  Windows

**Comment by danielkaradachki** [comment]
Occurs when moving the mouse through another parent before hovering a child with children.
Repro: https://plnkr.co/edit/fuj9a0Anx7rLgOG7MKjP?p=preview
To replicate:

1. Hover item1 and wait to be expanded.
2. Hover item1.1 and wait to be expanded.
3. Quickly move to item1.1.2 through item1.2

**Comment by ipeshev** [comment]
Fixed in v0.5.0-dev.201806220941

**Comment by tsvetomir** [comment]
Shipped in v0.5.0 of the `@progress/kendo-angular-menu` package.
